indico/htdocs/js/indico/jquery/rulelistwidget.js

```diff
@@ -0,0 +1,213 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Indico; if not, see <http://www.gnu.org/licenses/>.
+ */
+
+(function($) {
+    'use strict';
+
+    function _listValue(option) {
+        if (option === '') {
+            return [];
+        } else {
+            var parsedValue = parseInt(option, 10);
+            return [isNaN(parsedValue) ? option : parsedValue];
+        }
+    }
+
+    function _optionValue(option) {
+        if (option.length === 0) {
+            return '';
+        } else {
+            return option[0];
+        }
+    }
+
+    $.widget('indico.rulelistwidget', {
+        options: {
+        },
+
+        _renderRuleCondition: function(ruleContext, condition) {
+            var self = this;
+
+            var $select = $('<select>')
+                .append(_.map(condition.options, function(elem) {
+                    var optionTitle = elem[1];
+                    var id = elem[0];
+                    return $('<option>', {
+                        value: id
+                    }).text(optionTitle);
+                })).on('change', function() {
+                    self._optionSelected(ruleContext, condition, $(this).val());
+                    self.element.trigger('change');
+                });
+
+            if (condition.required) {
+                $select.prepend($('<option disabled selected>', {
+                    value: ''
+                }).text($T.gettext('Select an option')));
+            }
+
+            var $html = $('<div>', {
+                class: 'rule-condition'
+            }).attr('data-condition', condition.value);
+
+            if (condition.labelText) {
+                $html.append($('<label>').text(condition.labelText));
+            }
+            return $html.append($select);
+        },
+
+        _optionSelected: function(ruleContext, condition, newOption) {
+            ruleContext.rule[condition.value] = _listValue(newOption);
+            this._checkConditionIncompatibility(ruleContext);
+            this.element.trigger('change');
+        },
+
+        _onChange: function() {
+            // Simplify rule data by remove "field: any"
+
+            this.ruleData.forEach(function(rule) {
+                _.keys(rule).forEach(function(k) {
+                    if (rule[k] === '*') {
+                        delete rule[k];
+                    }
+                });
+            });
+
+            this.element.val(JSON.stringify(this.ruleData));
+        },
+
+        _checkConditionIncompatibility: function(ruleContext) {
+            ruleContext.$element.find('.rule-condition').show();
+
+            this.conditionSpec.forEach(function(condition) {
+                var $condition = ruleContext.$element.find('.rule-condition[data-condition=' + condition.value + ']');
```

Comment:
I agree that it would be better to do as you say, but why would we have a
value (a parameter we can control) contain that character?

On Wed, 7 Sep 2016, 16:49 Adrian, <notifications@github.com> wrote:

> In indico/htdocs/js/indico/jquery/rulelistwidget.js
> <https://github.com/indico/indico/pull/2595#discussion_r77837393>:
>
> > +            this.ruleData.forEach(function(rule) {
> > +                _.keys(rule).forEach(function(k) {
> > +                    if (rule[k] === '*') {
> > +                        delete rule[k];
> > +                    }
> > +                });
> > +            });
> > +
> > +            this.element.val(JSON.stringify(this.ruleData));
> > +        },
> > +
> > +        _checkConditionIncompatibility: function(ruleContext) {
> > +            ruleContext.$element.find('.rule-condition').show();
> > +
> > +            this.conditionSpec.forEach(function(condition) {
> > +                var $condition = ruleContext.$element.find('.rule-condition[data-condition=' + condition.value + ']');
>
> now try a value containing ] ;) I'd rather use
> .find('.rule-condition').filter() with a callback that checks the
> condition
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/indico/indico/pull/2595/files/a503f55dc6f738b72d5a14a8bee21c03f762ba26#r77837393>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAAKi9tHkZt4BdOpME1duIzTdQOIgp6Aks5qns7ugaJpZM4J26K0>
> .
>